source/faucet/main.d

```diff
@@ -118,12 +117,36 @@ private auto splitTx (UR) (UR utxo_rng, uint count)
             );
 }
 
+/*******************************************************************************
+
+    Merges the Outputs from `utxo_rng` into a range of transactions
+    with a single input and output.
+
+    Params:
+      UR = Range of tuple with an `Output` (`value`) and
+            a `Hash` (`key`), as its first and second element, respectively
+
+    Returns:
+      A range of Transactions
+
+*******************************************************************************/
+
+private auto mergeTx (UR) (UR utxo_rng) @safe
+{
+    static assert (isInputRange!UR);
+
+    return utxo_rng
+            .filter!(tup => tup.value.output.value > Amount(count))
```

Comment:
And this doesn't look right either. The filter was a workaround against an issue we saw were we created too small UTXOs. Merging by definition doesn't have that issue.

Reply:
Yes, I moved the workaround  from `splitTx` to here, but this eventually needs to be replaced.
I will have to come up with a way to call merge when there exists an `Amount` that is less than `count`.

Reply:
I still don't understand: Merge doesn't have the bug, does it ? It's only `split` that has it.

Reply:
If `split` creating `Amount`s less than `count` is considered a bug, then yes.
How can we prevent `split` from creating these "remainders" when it's called?

Reply:
> If split creating Amounts less than count is considered a bug, then yes.

Yes. So why was the code removed from `split`, and put in `merge` ?

> How can we prevent `split` from creating these "remainders" when it's called?

I don't think the remainders are a problems. I actually think the filter  call was legit.

Reply:
> Yes. So why was the code removed from split, and put in merge ?

I remember, I was thinking of sending these remainders to an invalid address instead of filtering, during the `merge` process. The `filter` there should have been temporary.

But if the filter call in `split` is legit, then it works very well.